src/Components/Components/src/ComponentBase.cs

```diff
@@ -175,22 +190,22 @@ void IComponent.Configure(RenderHandle renderHandle)
         /// Method invoked to apply initial or updated parameters to the component.
         /// </summary>
         /// <param name="parameters">The parameters to apply.</param>
-        public virtual Task SetParametersAsync(ParameterCollection parameters)
+        Task IComponent.SetParametersAsync(ParameterCollection parameters)
```

Comment:
Our idea of making this explicit with *overrides* is that you can't use the `base` keyword within in an explicit interface implementation. So it's not really possible to interact with by *overriding* or it would require a lot more design.

It seemed more natural to give you a before/after lifecycle model. The only use case I can really think of for before is to set default values, but it's otherwise SUPER painful to do.